File: analytics-sdk/analytics-storage-server/src/middleware/validateRequest.js
const ApiError = require('../errors/ApiError');

function validateRequest(requiredFields = []) {
  return (req, res, next) => {
    const body = req.body || {};
    const missingFields = requiredFields.filter(field => !(field in body));
    if (missingFields.length > 0) {
      return next(new ApiError(400, `Missing fields: ${missingFields.join(', ')}`));
    }
    next();
  };
}

module.exports = validateRequest;

File: analytics-sdk/analytics-storage-server/src/middlewares/validate_request.py
from fastapi import Request
from src.errors.api_error import ApiError

async def validate_request(request: Request, required_fields: list):
    body = await request.json()
    missing_fields = [field for field in required_fields if field not in body]
    if missing_fields:
        raise ApiError(400, f"Missing fields: {', '.join(missing_fields)}")